lib/tools/adb-emu-commands.js

```diff
@@ -38,9 +38,24 @@ emuMethods.fingerprint = async function (fingerprintId, udid = undefined) {
   if (udid) {
     await this.setDeviceId(udid);
   }
+  this.resetTelnetAuthToken();
   await this.adbExec(['emu', 'finger', 'touch', fingerprintId]);
 };
 
+emuMethods.rotate = async function (udid) {
+  if (!_.isUndefined(udid) && !(await this.isEmulatorConnected(udid)) ) {
```

Comment:
probably it makes sense to refactor the stuff and move setting of the id/token reset into separate method. Also, there is no need to call the stuff twice, so we could also cache results and avoid calling the save some time if an emulator command is invoked more than once.